include/oneapi/dpl/pstl/utils.h

```diff
@@ -784,6 +785,32 @@ union __lazy_ctor_storage
     }
 };
 
+// Utility to explicitly call the destructor of __lazy_ctor_storage as a callback functor
+struct __lazy_ctor_storage_deleter
```

Comment:
I believe it is a callable deleter for `__lazy_ctor_storage` which is storage that has a delayed "lazy" constructor.  Perhaps it would be better to instead add a static member function to the `__lazy_ctor_storage` union, `get_deleter_callable()`, which returns a lambda to delete a `__lazy_ctor_storage` passed as an argument.  This would remove any confusion, and group these together.